pkg/controller/controller_utils.go

```diff
@@ -1064,7 +1064,8 @@ func AddOrUpdateTaintOnNode(ctx context.Context, c clientset.Interface, nodeName
 		if !updated {
 			return nil
 		}
-		return PatchNodeTaints(ctx, c, nodeName, oldNode, newNode)
+		_, err = c.CoreV1().Nodes().Update(ctx, newNode, metav1.UpdateOptions{})
```

Comment:
For historical reasons there is a copy of `AddOrUpdateTaintOnNode` [here](https://github.com/jprzychodzen/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/node/helpers/taints.go#L51), should that also be updated or does only updating KCM fix the race condition?

Reply:
This specific race condition happens because of KCM.

I prefer to have preemptive fix in other PR, as it's more a potential bug without materialized impact.

Reply:
in general, we try to avoid get/modify/write of the Node object because it means this will drop any Node fields unknown to this binary if it is linked to an older Node type definition

Reply:
that's because the patch still applies cleanly (does not conflict) even on top of the newly updated fields